python/social_bot/tasks.py

```diff
@@ -631,6 +754,7 @@ def task_specific_observation(self, agent):
             joints_states = agent.get_internal_states()
```

Comment:
Remove the "else:" part
The only diff from "use full world states" is joints_states that contains angles and velocity of two wheels. Essentially another representation of position changes, velocity, yaw and yaw rate that are already in agent_vel. 
Sine we have the _use_full_states part, we should remove these. Option _use_full_states should also be removed as we dont need it anymore.

Reply:
Since this code behavior has been in the code base for some time, I can't be sure that nobody is depending on this behavior.  I'm actually more comfortable leaving the else in, and adding this change in with flag protection.

Reply:
It's fine to remove them, we don't need 2 similar implementation. Let's keep the code clean. 
 : )